compiler/vmops.nim

```diff
@@ -102,6 +102,37 @@ proc staticWalkDirImpl(path: string, relative: bool): PNode =
     result.add newTree(nkTupleConstr, newIntNode(nkIntLit, k.ord),
                               newStrNode(nkStrLit, f))
 
+proc compileSettingImpl(conf: ConfigRef, switch: string): string =
+  case switch.normalize
+  of "arguments": result = conf.arguments
+  of "outfile": result = conf.outFile.string
+  of "outdir": result = conf.outDir.string
+  of "nimcachedir": result = conf.nimcacheDir.string
+  of "projectname": result = conf.projectName
+  of "projectpath": result = conf.projectPath.string
+  of "projectfull": result = conf.projectFull.string
+  of "command": result = conf.command
+  of "commandline": result = conf.commandLine
+  of "linkoptions": result = conf.linkOptions
+  of "compileoptions": result = conf.compileOptions
+  of "ccompilerpath": result = conf.cCompilerPath
+  else: discard
```

Comment:
`else: ` needs to crash, via `doAssert false` or even `globalError(conf, "invalid: " & switch)`
otherwise user may silently get empty results instead of an error when misspelling a field, or when a field is later added and user calls `compileSetting("newfield")` which works on new nim versions but silently gives wrong results on older nim's instead of giving an error.

Silent wrong results is the worst behavior, making bugs hard to find.


ditto with compileSettingSeqImpl

Reply:
Done